dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/CosmosSystemTextJSonSerializer.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+// Taken from https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4332
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Cosmos;
+
+/// <summary>
+/// This class provides a default implementation of System.Text.Json Cosmos Linq Serializer.
+/// </summary>
+public class CosmosSystemTextJsonSerializer : CosmosLinqSerializer
+{
+    /// <summary>
+    /// A read-only instance of <see cref="JsonSerializerOptions"/>.
+    /// </summary>
+    private readonly JsonSerializerOptions _jsonSerializerOptions;
+
+    /// <summary>
+    /// Creates an instance of <see cref="CosmosSystemTextJsonSerializer"/>
+    /// with the default values for the Cosmos SDK
+    /// </summary>
+    /// <param name="jsonSerializerOptions">An instance of <see cref="JsonSerializerOptions"/> containing the json serialization options.</param>
+    public CosmosSystemTextJsonSerializer(
+        JsonSerializerOptions jsonSerializerOptions)
+    {
+        this._jsonSerializerOptions = jsonSerializerOptions;
+    }
+
+    /// <inheritdoc/>
+    public override T FromStream<T>(Stream stream)
+    {
+        if (stream == null)
+        {
+            throw new ArgumentNullException(nameof(stream));
+        }
+
+        if (stream.CanSeek && stream.Length == 0)
+        {
+            return default;
+        }
+
+        if (typeof(Stream).IsAssignableFrom(typeof(T)))
+        {
+            return (T)(object)stream;
+        }
+
+        using (stream)
+        {
+            using StreamReader reader = new(stream);
+            return JsonSerializer.Deserialize<T>(reader.ReadToEnd(), this._jsonSerializerOptions);
```

Comment:
JsonSerializer.Deserializer has overloads that can deserialize directly from Stream. Can those be used instead of reading the whole contents into a string first?

Reply:
Apparently the version of STJ our SDK depends on is too old, according to the discussion at https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4332/files#r1562040906.

I could address that by adding an explicit reference to a newer STJ here in this project if that makes sense to you?

Reply:
Ouch, STJ 4.6.0 is 5 years. The rest of the SK assemblies reference the 8.0 version:
https://github.com/microsoft/semantic-kernel/blob/0c89e0bd4314b4f4c913563258ffefadedab1afe/dotnet/Directory.Packages.props#L37
and since this assembly depends on some of those, I expect you're already being forced up. Might as well benefit from it :smile:

Reply:
Yep, comes through the abstractions project.  Updated.